### ISSUE TITLE
smart refresh enhancement (timeout refresh)

### ISSUE THREAD

**Issue by kyleshay** [DestinyItemManager/DIM#203]
Instead of refreshing on a period of a set time, it may be better to instead detect if a user has been idle on their device for a given amount of time. When the page becomes active again, a refresh can either happen, or a non-blocking prompt can request permission from the user to refresh the inventory with a dialog explaining that the user 'has been idle for XX minutes and that DIM would like to automatically refresh their items'.

Basically:
If a user is idle from their phone for > 30 seconds flip a boolean that will trigger a refresh when the user comes back.

This library does this exact thing pretty well:
http://serkanyersen.github.io/ifvisible.js/demo.html

I'm sure there are others. We can change the timeout times as well.

**Comment by SunburnedGoose** [comment]
+1

**Comment by bhollis** [comment]
Implemented via #299.
